clients/lsp-clangd.el

```diff
@@ -235,5 +235,24 @@ returned to avoid that the echo area grows uncomfortably."
         (car (s-lines (lsp--render-element contents)))))))
 
 
+(defun lsp-clangd-to-other ()
+  "Open the corresponding header/source file."
+  (interactive)
+  (if-let ((other-fname
+            ;; find-workspace might be too brittle if users make their own lsp IDs
+            ;; eg. clangd-remote
+            (with-lsp-workspace (lsp-find-workspace 'clangd)
+              ;; TODO define an extension with lsp-interace
+              ;; similar to rust-analyzer on lines 278-292 in
+              ;; lsp-protocol.el
+              (lsp-send-request (lsp-make-request
+                                 "textDocument/switchSourceHeader"
+                                 (lsp--text-document-identifier))))
+            ))
+      ;; TODO maybe worth using lsp-goto-location instead
+      (find-file (lsp--uri-to-path other-fname))
+    (lsp--info "This file doesn't have a corresponding file")
+    nil))
```

Comment:
Is this needed?

Reply:
By "this" are you referring you to the `nil` return value or lsp--info message?

Reply:
also, lsp-goto-location doesn't show up when I grep for it in the whole repo, which suggests I can get rid of the TODO?

Reply:
Yes, the "this"  points to `nil`.
Also since the server is returning URI instead of location, I think you can remove the TODO.